hybrid/pkg/agent/agent_interceptor.go

```diff
@@ -0,0 +1,93 @@
+package hybrid_agent
+
+import (
+	"context"
+	"encoding/json"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hewlettpackard/hybrid/pkg/common"
+	nodeattestorv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/nodeattestor/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type AgentInterceptorInterface interface {
+	Recv() (*nodeattestorv1.Challenge, error)
+	Send(challenge *nodeattestorv1.PayloadOrChallengeResponse) error
```

Comment:
I did override them

Reply:
It's an unecessary risk to take, because we must obey the interfaces, which is good, because if they change we would know and it'll be easier to act on our side to keep up with those changes